From 6a029459fe1656e3082a64958191b8e84021995b Mon Sep 17 00:00:00 2001
From: Christophe Kerello <christophe.kerello@foss.st.com>
Date: Thu, 7 Mar 2024 17:57:23 +0100
Subject: [PATCH 0716/1695] drivers: stm32_omm: always clock OSPI instances if
 mux is enabled

If the mux is enabled, the 2 OSPI clocks have to be always enabled.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 546497e283c03f1af0501972e186eee0fdb7eb50 drivers: stm32_omm: always clock OSPI instances if mux is enabled]
Change-Id: Id94b60bf5a0822321a0e2ca93e75be7654812131
Signed-off-by: Christophe Kerello <christophe.kerello@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/optee/optee_os/+/366667
Reviewed-by: Christophe KERELLO <christophe.kerello@st.com>
Reviewed-by: Gatien CHEVALLIER <gatien.chevallier@st.com>
Domain-Review: Christophe KERELLO <christophe.kerello@st.com>
Tested-by: Christophe KERELLO <christophe.kerello@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 core/drivers/stm32_omm.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/core/drivers/stm32_omm.c b/core/drivers/stm32_omm.c
index 2afebc5b5..d3388067f 100644
--- a/core/drivers/stm32_omm.c
+++ b/core/drivers/stm32_omm.c
@@ -315,6 +315,19 @@ static void stm32_omm_configure(void)
 			req2ack | cssel_ovr | omm_d->mux);
 
 	clk_disable(omm_d->clock);
+
+	if (omm_d->mux & _OCTOSPIM_CR_MUXEN) {
+		/*
+		 * If the mux is enabled, the 2 OSPI clocks have to be
+		 * always enabled
+		 */
+		for (ospi_i = 0; ospi_i < OSPI_NB; ospi_i++) {
+			ospi_d = &omm_d->ospi_d[ospi_i];
+
+			if (clk_enable(ospi_d->clock))
+				panic();
+		}
+	}
 }
 
 static void stm32_omm_setup(void)
-- 
2.47.2

